src/rpc/methods/eth.rs

```diff
@@ -1669,26 +1669,26 @@ impl RpcMethod<0> for EthSyncing {
         ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
         (): Self::Params,
     ) -> Result<Self::Ok, ServerError> {
-        let crate::rpc::sync::RPCSyncState { active_syncs } =
-            crate::rpc::sync::SyncState::handle(ctx, ()).await?;
-        match active_syncs
-            .into_iter()
-            .rev()
-            .find_or_first(|ss| ss.stage() != SyncStage::Idle)
-        {
-            Some(sync_state) => match (sync_state.base(), sync_state.target()) {
-                (Some(base), Some(target)) => Ok(EthSyncingResult {
-                    done_sync: sync_state.stage() == SyncStage::Complete,
-                    current_block: sync_state.epoch(),
-                    starting_block: base.epoch(),
-                    highest_block: target.epoch(),
-                }),
-                _ => Err(ServerError::internal_error(
-                    "missing syncing information, try again",
-                    None,
-                )),
-            },
-            None => Err(ServerError::internal_error("sync state not found", None)),
+        let sync_status: crate::chain_sync::SyncStatusReport =
+            crate::rpc::sync::SyncStatus::handle(ctx, ()).await?;
+        match sync_status.get_status() == NodeSyncStatus::Syncing {
+            true => {
```

Comment:
when matching on boolean values, I feel `If and Else` is more suitable and readable

Reply:
In general, I agree, though we already have several occurrences of such style in Forest, so it's not a huge deal. For this particular one, I'd avoid comparing and matching on `sync_status.get_status()` directly. This way, when a new state is introduced, we'll get a compilation error due to an unhandled enum variant.